src/zkevm_specs/evm/execution/callop.py

```diff
@@ -16,18 +16,26 @@
 from ..precompiled import PrecompiledAddress
 
 
-def call_staticcall(instruction: Instruction):
+def callop(instruction: Instruction):
     opcode = instruction.opcode_lookup(True)
-    is_call, _ = instruction.pair_select(opcode, Opcode.CALL, Opcode.STATICCALL)
+    is_call, is_delegatecall, is_staticcall = instruction.multiple_select(
+        opcode, (Opcode.CALL, Opcode.DELEGATECALL, Opcode.STATICCALL)
+    )
     instruction.responsible_opcode_lookup(opcode)
 
     callee_call_id = instruction.curr.rw_counter
 
     tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
     reversion_info = instruction.reversion_info()
     caller_address = instruction.call_context_lookup(CallContextFieldTag.CalleeAddress)
+    parent_caller_address = instruction.select(
+        is_delegatecall, instruction.call_context_lookup(CallContextFieldTag.CallerAddress), FQ(0)
+    )
```

Comment:
`instruction.select` always computes the inner arguments first and only then selects the appropriate one.

So in case `is_delegatecall == false` it will still add a lookup to the call context. Not that this is a bug, but we should be careful this doesn't lead to test errors when the case is not DELEGATECALL.

Reply:
Although in case of call context, it may not be an issue because the call's context is always populated? Correct me if I'm wrong.

Reply:
Yes. I see. I added `2` to `rw_counter` [on this line](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/294/files/209d7ba6ca18b2a705b6959c9fcfa048d17fbe38#diff-10487489ead1ac341b7671be200b7c7b4b63c000a33f05ffb0bcfe63e73d6ff0L142-R152) even if they both are not used for `CALL` and `STATICCALL` opcodes.

I suppose if I could remove `instruction.select` and fix as below. Since the both values are used in the next call's context setup.
```
# Fixed as:
parent_caller_address = instruction.call_context_lookup(CallContextFieldTag.CallerAddress)
parent_call_value = instruction.call_context_lookup(CallContextFieldTag.Value)

...

# Setup next call's context.
for (field_tag, expected_value) in [
    ...
    (
        CallContextFieldTag.CallerAddress,
        instruction.select(
            is_delegatecall, parent_caller_address.expr(), caller_address.expr()
        ),
    ),
    ...
    (
        CallContextFieldTag.Value,
        instruction.select(is_delegatecall, parent_call_value.expr(), value.expr()),
    ),
    ...
]:
    instruction.constrain_equal(
        instruction.call_context_lookup(field_tag, call_id=callee_call_id),
        expected_value,
    )
```

@roynalnaruto WDYT? Thanks.

Reply:
But in that case, would the `parent_call_value` be `FQ.zero()` if it's not a delegatecall?

Reply:
@silathdiir This change should be fine, as `parent_caller_address` and `parent_call_value` only get used if `is_delegatecall == true`

Reply:
Fixed as above comment. Thanks.